### PR TITLE
Implement call activity props

### LINKED ISSUE #55: Call Activity props
* [x] Called element (none / BPMN / CMMN)
* [x] Business Key
* [x] Business Key Expression
* [x] Delegate Variable Mapping (if BPMN is called)

### PR DESCRIPTION
**Pull request by barmac**
Closes #55

### CODE REVIEW

File: src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js [hunk 1]
```diff
@@ -188,18 +189,13 @@ function ScriptGroup(element) {
...
-// @TODO: implement, hide with no entries in the meantime
 function CallActivityGroup(element) {
 
-  if (!is(element, 'bpmn:CallActivity')) {
-    return null;
-  }
-
   const group = {
-    label: 'Call Activity',
+    label: 'Called Element',
```
Comment (MaxTru) [inline src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js:195]: According to https://docs.google.com/document/d/1Tcw1GSemSFFSXZDlfGEygvW4RHPInQnC4cLuAQXlQ6U/edit#heading=h.ivjzwe7sms6b, should this not rather be in sentence case: `Called element`? 

cc. @andreasgeier

File: src/provider/camunda-platform/properties/CallActivityProps.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    const businessObject = getBusinessObject(element);
+    let extensionElements = businessObject.get('extensionElements');
+
+    // (1) If there are no extension elements, create camunda:In and update element's properties
+    if (!extensionElements) {
+      extensionElements = createElement('bpmn:ExtensionElements', {}, businessObject, bpmnFactory);
+      const businessKeyItem = createBusinessKey(extensionElements);
+      extensionElements.set('values', [ businessKeyItem ]);
+
+      modeling.updateProperties(element, { extensionElements });
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/CallActivityProps.js:172]: We don't we fire a specific `properties-panel.update-XYZ' command here?

File: src/provider/camunda-platform/properties/CallActivityProps.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  const setValue = (value) => {
+    const properties = { ...DEFAULT_PROPS };
+
+    if (value === 'bpmn') {
+      properties['calledElement'] = '';
+    } else if (value === 'cmmn') {
+      properties['camunda:caseRef'] = '';
+    }
+
+    commandStack.execute('element.updateProperties', { element, properties });
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/CallActivityProps.js:101]: Any reason we don't fire the specific 'properties-panel.update-businessobject' command?

File: src/provider/camunda-platform/properties/CalledBpmnProps.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  };
+
+  const setValue = value => {
+    modeling.updateProperties(element, { calledElement: value || '' });
+  };
+
+  return <TextField
+    element={ element }
+    id="calledElement"
+    label={ translate('Called Element') }
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/CalledBpmnProps.js:78]: same as before, sentence case?

File: src/provider/camunda-platform/properties/CalledBpmnProps.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  const modeling = useService('modeling');
+  const translate = useService('translate');
+  const debounce = useService('debounceInput');
+
+  const getValue = () => {
+    return getBusinessObject(element).get('camunda:calledElementVersion');
+  };
+
+  const setValue = value => {
+    modeling.updateProperties(element, { calledElementVersion: value });
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/CalledBpmnProps.js:135]: same question as before, why not use the properties-panel namespaced command?

File: src/provider/camunda-platform/properties/CalledBpmnProps.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+  };
+
+  const setValue = value => {
+    modeling.updateProperties(element, { calledElementVersionTag: value });
+  };
+
+  return <TextField
+    element={ element }
+    id="calledElementVersionTag"
+    label={ translate('Version Tag') }
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/CalledBpmnProps.js:166]: sentence cased?

File: src/provider/camunda-platform/properties/CalledCmmnProps.js [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  };
+
+  const setValue = value => {
+    modeling.updateProperties(element, { caseRef: value || '' });
+  };
+
+  return <TextField
+    element={ element }
+    id="calledElementCaseRef"
+    label={ translate('Case Ref') }
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/CalledCmmnProps.js:68]: sentence cased?

File: test/spec/provider/camunda-platform/CallActivityProps.spec.js [hunk 8]
```diff
@@ -0,0 +1,804 @@
...
+      await act(() => {
+        selection.select(element);
+      });
+
+      // when
+      const input = domQuery(':is(input, select)[name=calledElementBusinessKeyExpression]', container);
+      changeInput(input, 'newValue');
+
+      // then
+      expect(
```
Comment (MaxTru) [inline test/spec/provider/camunda-platform/CallActivityProps.spec.js:240]: split on three lines here is inconsistent to the rest of the file.

File: test/spec/provider/camunda-platform/CallActivityProps.spec.js [hunk 9]
```diff
@@ -0,0 +1,804 @@
...
+
+        // then
+        expect(getBusinessKey(element)).to.eql('#{execution.processBusinessKey}');
+        expect(
+          getBusinessObject(element).get('extensionElements').get('values')).to.have.lengthOf(2);
+      })
+    );
+
+
+    it('should remove extension elements if none is left',
```
Comment (MaxTru) [inline test/spec/provider/camunda-platform/CallActivityProps.spec.js:314]: Maybe add as testCase: `should re-use existing extension elements`

File: test/spec/provider/camunda-platform/CallActivityProps.spec.js [hunk 10]
```diff
@@ -0,0 +1,804 @@
...
+
+
+    it('should update on external change', shouldUpdateOnExternalChange(
+      'BPMN_DelegateVariableMappingDelegateExpression', input, property
+    ));
+  });
+
+
+
+  function shouldDisplay(elementId, inputName, propertyName) {
```
Comment (MaxTru) [inline test/spec/provider/camunda-platform/CallActivityProps.spec.js:708]: :+1: :rocket:

**Review by MaxTru — COMMENTED** [review]
What a beast!

I cannot find any issue functionality-wise, it just works.

I only have detail comments.

File: src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js (hunk 1 above)
Comment (andreasgeier) [reply src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js:195]: I don't want to enforce new wording rules in the middle of the implementation. If you like, you can correct this immediately, but later in the design finishing is also fine.

**Comment by andreasgeier** [comment]
Missing job configuration as section

**Comment by andreasgeier** [comment]
Should we really move CMMN as a target to the new panel? We do not support CMMN anymore.

**Comment by barmac** [comment]
CMMN is still documented though: https://docs.camunda.org/manual/latest/reference/bpmn20/subprocesses/call-activity/#create-a-case-instance

**Comment by andreasgeier** [comment]
> Should we really move CMMN as a target to the new panel?

Double checked with PM. We keep it.

**Review by andreasgeier — APPROVED** [review]
✅ &nbsp; LGTM

File: src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js (hunk 1 above)
Comment (barmac) [reply src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js:195]: OK I'll fix this.

File: src/provider/camunda-platform/properties/CallActivityProps.js (hunk 2 above)
Comment (barmac) [reply src/provider/camunda-platform/properties/CallActivityProps.js:172]: There is no need for this. The properties panel command is only a workaround for updating nested business object.

File: src/provider/camunda-platform/properties/CallActivityProps.js (hunk 3 above)
Comment (barmac) [reply src/provider/camunda-platform/properties/CallActivityProps.js:101]: Cf. https://github.com/bpmn-io/bpmn-properties-panel/pull/66#discussion_r689386494

File: src/provider/camunda-platform/properties/CalledBpmnProps.js (hunk 5 above)
Comment (barmac) [reply src/provider/camunda-platform/properties/CalledBpmnProps.js:135]: Cf. https://github.com/bpmn-io/bpmn-properties-panel/pull/66#discussion_r689386494

File: src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js (hunk 1 above)
Comment (barmac) [reply src/provider/camunda-platform/CamundaPlatformPropertiesProvider.js:195]: BTW well done with the document @andreasgeier. It's super useful 👍

File: test/spec/provider/camunda-platform/CallActivityProps.spec.js (hunk 9 above)
Comment (barmac) [reply test/spec/provider/camunda-platform/CallActivityProps.spec.js:314]: I will add a check in [the test case below] to make sure the extension element is not removed.

**Review by MaxTru — APPROVED** [review]

**Pull request merged**